core/src/main/java/org/apache/iceberg/V2Metadata.java

```diff
@@ -278,37 +277,41 @@ static Types.StructType fileType(Types.StructType partitionType) {
         DataFile.REFERENCED_DATA_FILE);
   }
 
-  static class IndexedManifestEntry<F extends ContentFile<F>>
-      implements ManifestEntry<F>, IndexedRecord {
-    private final org.apache.avro.Schema avroSchema;
+  static class ManifestEntryWrapper<F extends ContentFile<F>>
+      implements ManifestEntry<F>, StructLike {
+    private final int size;
     private final Long commitSnapshotId;
-    private final IndexedDataFile<?> fileWrapper;
+    private final DataFileWrapper<?> fileWrapper;
     private ManifestEntry<F> wrapped = null;
 
-    IndexedManifestEntry(Long commitSnapshotId, Types.StructType partitionType) {
-      this.avroSchema = AvroSchemaUtil.convert(entrySchema(partitionType), "manifest_entry");
+    ManifestEntryWrapper(Long commitSnapshotId) {
+      this.size = entrySchema(Types.StructType.of()).columns().size();
```

Comment:
Same comment as above, just seems like we are doing a lot of work to get a constant

Reply:
I know what you mean, but I didn't want this to get out of sync with `entrySchema`. Still calling that method seemed like the safest path.